Index: mongodb/src/mongo/db/startup_warnings_common.cpp
===================================================================
--- mongodb.orig/src/mongo/db/startup_warnings_common.cpp
+++ mongodb/src/mongo/db/startup_warnings_common.cpp
@@ -95,7 +95,7 @@
     }
 #endif
 
-#if !defined(_WIN32)
+#if !defined(_WIN32) && !defined(DO_NOT_SHOW_IS_ROOT_WARNINGS)
     if (getuid() == 0) {
         log() << "** WARNING: You are running this process as the root user, "
               << "which is not recommended." << startupWarningsLog;
